src/catchup/CatchupWork.cpp

```diff
@@ -110,8 +111,17 @@ CatchupWork::downloadApplyBuckets()
     std::vector<std::string> hashes = has.differingBuckets(mLocalState);
     auto getBuckets = std::make_shared<DownloadBucketsWork>(
         mApp, mBuckets, hashes, *mDownloadDir);
+
+    // A consequence of FIRST_PROTOCOL_SHADOWS_REMOVED upgrade, inputs or
+    // outputs aren't published to the archives anymore: new-style merges are
+    // re-started from scratch To avoid going out of sync during online catchup,
```

Comment:
Missing period between "scratch" and "To".

Also on the next line: "application work wait" should be "application work to wait".